src/EllCrv/EllCrv.jl

```diff
@@ -46,12 +51,11 @@ export base_field, division_polynomialE, EllipticCurve, infinity,
 ################################################################################
 
 mutable struct EllCrv{T}
-  base_field::Ring
+  base_field::Field
```

Comment:
Was it `Ring` intentionally before? Just wondering as I know that e.g. "elliptic curves over Z/nZ not a prime are used in factorization.